tests/drivers/uart/uart_async_api/testcase.yaml

```diff
@@ -110,3 +110,8 @@ tests:
     extra_args:
       - DTC_OVERLAY_FILE="boards/nucleo_h723zg.overlay;boards/nucleo_h723zg_nocachemem.overlay"
       - EXTRA_CONF_FILE=stm32_nocache_mem_dt.conf
+  drivers.uart.async_api.uart_emul:
+    harness: ztest
+    platform_allow:
+      - qemu_x86
+      - native_posix
```

Comment:
Using native_sim instead of native_posix now. I also moved the devicetree overlays into one special for the uart emulation node, which is then used for both platforms.